contracts/staking/interfaces/IRewardCalculator.sol

```diff
@@ -7,10 +7,11 @@ pragma solidity 0.8.25;
 
 interface IRewardCalculator {
     function calculateReward(
-        uint64 stakeAmount,
+        uint256 stakeAmount,
         uint64 startTime,
         uint64 endTime,
         uint256 initialSupply,
-        uint256 endSupply
-    ) external returns (uint256);
+        uint256 endSupply,
+        uint64 rewardBasisPoints
```

Comment:
I think the curve definition should be the left to implementations, and not be in the interface. The interface should be kept to inputs needed to calculate a point on the curve itself.

Reply:
random thought, would it make sense to take in arbitrary bytes as input as well for output, to make it open to any implemetnation

Reply:
I think @cam-schultz may have brought this up previously. I don't think it makes sense because the caller and implementation would still need to agree on the expected data format (i.e. what the parameters are), and changes to that could just as easily be made as changes to the interface parameters themselves.

Reply:
I think in the conversation you're referencing I was suggesting providing arbitrary bytes to the rewards curve constructor, but having the rewards curve be a standalone contract from the staking manager acheives the same generality. In either case, yes, we need an agreed upon interface.

Reply:
I agree that the validator manager and the reward manager still need to have the same expectations of format, but take in bytes makes it so that reward managers that take into different inputs can still fall under the same interface. Right now if if a reward manager were to take a different input, it'd need to define a separate interface

Reply:
@minghinmatthewlam I think that having a bytes interface would actually be coutner-productive. In order for the staking contract and reward calculator to work together, they would both have to change their implementations. If they're both changing their implementations, it would be easy to change the interface between them, and I think it would be a lot clearer, where a bytes interface obfuscates the fact that the contracts might not actually agree what form the parameters should take